eth_defi/aave_v3/loan.py

```diff
@@ -0,0 +1,125 @@
+"""Aave v3 deposit."""
+from typing import Callable
+
+from web3 import Web3
+
+from eth_defi.abi import get_abi_by_filename
+from eth_defi.hotwallet import HotWallet
+
+
+def approve_token(web3: Web3, token_address: str, spender: str, amount: int) -> Callable:
+    """
+    Approve a specified amount of token to be spent.
+
+    Example:
+
+    .. code-block:: python
+
+        # build transaction to approve ERC-20 to be spent.
+        approve_func = approve_token(
+            web3=web3,
+            token_address=token_address,
+            spender=spender,
+            amount=amount
+        )
+
+        tx = approve_func.build_transaction(
+            {
+                "from": hot_wallet_address,
+                "chainId": web3.eth.chain_id
+            }
+        )
+        tx = fill_nonce(web3, tx)
+
+        # sign and broadcast
+        signed = hot_wallet_account.sign_transaction(tx)
+        tx_hash = web3.eth.send_raw_transaction(signed.rawTransaction)
+        receipt = web3.eth.get_transaction_receipt(tx_hash)
+        assert receipt.status == 1, "USDC transfer reverted"
+
+    :param web3:
+        Web3 Instance
+    :param token_address:
+        The address of the Aave v3 reserve token that you want to deposit.
+    :param spender:
+        The address of the spender to be approved.
+    :param amount:
+        The amount of token to be approved.
+    :return:
+        Callable approve transaction.
+    """
+
+    # Load token contract ABI
+    token_abi = get_abi_by_filename("ERC20MockDecimals.json")["abi"]
+
+    # Get token contract instance
+    token_contract = web3.eth.contract(address=token_address, abi=token_abi)
+
+    # Approve the token contract to spend the specified amount of token
+    transaction = token_contract.functions.approve(spender, amount)
+
+    # Return the transaction
+    return transaction
```

Comment:
IMO I don't think we would want to split this approve token process to a separate function, it should either be included in supply process or user can do it himself